com.unity.perception/Runtime/GroundTruth/Labeling/LabelManager.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System.Collections.Generic;
+using UnityEngine.Perception.Randomization.Randomizers;
+
+namespace UnityEngine.Perception.GroundTruth
+{
+    /// <summary>
+    /// Manages the registration of <see cref="Labeling"/> components
+    /// </summary>
+    public class LabelManager
+    {
+        /// <summary>
+        /// Returns the active LabeledObjectsManager instance
+        /// </summary>
+        public static LabelManager singleton { get; } = new LabelManager();
+
+        const uint k_StartingIndex = 1;
+        uint m_CurrentObjectIndex = k_StartingIndex;
+        List<IGroundTruthGenerator> m_ActiveGenerators = new List<IGroundTruthGenerator>();
+        LinkedHashSet<Labeling> m_LabelsPendingRegistration = new LinkedHashSet<Labeling>();
+        LinkedHashSet<Labeling> m_RegisteredLabels = new LinkedHashSet<Labeling>();
+
+        /// <summary>
+        /// Returns the set of registered Labeling components
+        /// </summary>
+        public IEnumerable<Labeling> registeredLabels => m_RegisteredLabels;
+
+        /// <summary>
+        /// Registers all pending labels.
+        /// Called once per frame during LateUpdate by the <see cref="PerceptionUpdater"/>.
+        /// </summary>
+        public void RegisterPendingLabels()
+        {
+            if (m_RegisteredLabels.Count == 0)
+                m_CurrentObjectIndex = k_StartingIndex;
```

Comment:
'NextObjectIndex' is the standard way of naming a value like this. Why 'current'?

Reply:
This was copy pasted code from the old system, so I didn't think to hard about changing the naming conventions. I agree and will change this. Good catch.